lxd/db/profiles.go

```diff
@@ -173,7 +176,7 @@ DELETE FROM profiles_devices_config WHERE profile_device_id NOT IN (SELECT id FR
 
 // ExpandInstanceConfig expands the given instance config with the config
 // values of the given profiles.
-func ExpandInstanceConfig(config map[string]string, profiles []api.Profile) map[string]string {
+func ExpandInstanceConfig(globalConfig clusterIfaces.ClusterConfig, instType *instancetype.Type, config map[string]string, profiles []api.Profile) map[string]string {
```

Comment:
Can we move ExpandInstanceConfig and ExpabdInstanceDevices into either `lxd/instance` or `lxd/instance/instancetype` packages to avoid the need for the new interface?

Reply:
Because `profiles` could already contain `migration.statefule` and it would be applied to containers and VMs, we dont need instType here, when expanding from global server config too.